src/kv/kv.h

```diff
@@ -1495,80 +1498,106 @@ namespace kv
       bool public_only = false,
       Term* term = nullptr) override
     {
-      // This will return FAILED if the serialised transaction is being
-      // applied out of order.
-      // Processing transactions locally and also deserialising to the
-      // same store will result in a store version mismatch and
-      // deserialisation will then fail.
+      frame::FlatbufferDeserialiser fbd(data.data());
+      auto frames = fbd.get_frames();
+      Version v;
+      OrderedViews<S, D> views;
+      bool first_serialiser = true;
       auto e = get_encryptor();
-      D d(
-        e,
-        public_only ? kv::SecurityDomain::PUBLIC :
-                      std::optional<kv::SecurityDomain>());
-      if (!d.init(data))
+
+      for (auto& [frame, size] : frames)
       {
-        LOG_FAIL_FMT("Initialisation of deserialise object failed");
-        return DeserialiseSuccess::FAILED;
-      }
+        if (size > 0)
+        {
+          // This will return FAILED if the serialised transaction is being
+          // applied out of order.
+          // Processing transactions locally and also deserialising to the
+          // same store will result in a store version mismatch and
+          // deserialisation will then fail.
+          D d(
+            e,
+            public_only ? kv::SecurityDomain::PUBLIC :
+                          std::optional<kv::SecurityDomain>());
+
+          if (!d.init(frame, size))
+          {
+            LOG_FAIL_FMT("Initialisation of deserialise object failed");
+            return DeserialiseSuccess::FAILED;
+          }
 
-      Version v = d.template deserialise_version<Version>();
-      LOG_DEBUG_FMT("Deserialising {}", v);
+          if (first_serialiser)
```

Comment:
we are using two deserialisers now (one for replicated tables one for derived) so w only need to `rollback(v - 1)` once. But it's good to check that the version is consistent between the deserialisers